xonsh/commands_cache.py

```diff
@@ -99,11 +99,11 @@ def all_commands(self):
             # entries at the back.
             for cmd in executables_in(path):
                 key = cmd.upper() if ON_WINDOWS else cmd
-                allcmds[key] = (os.path.join(path, cmd), cmd in alss)
+                allcmds[key] = (os.path.join(path, cmd), alss.get(key, None))
         for cmd in alss:
             if cmd not in allcmds:
                 key = cmd.upper() if ON_WINDOWS else cmd
-                allcmds[key] = (cmd, True)
+                allcmds[key] = (cmd, None)
```

Comment:
But on line 183 you check: 
```
(path, alias) = self._cmds_cache[cached]
             if not alias or ignore_alias:
...
```
Which would evaluate to True since `alias` would be `None`.